standard/interfaces.md

```diff
@@ -242,24 +255,133 @@ If a `new` modifier is included in a declaration that doesn’t hide an inherite
+
+*constant_declaration* is described in [§15.4](classes.md#154-constants).
+
+As a *constant_declaration* is considered to have a default implementation ([§18.1](interfaces.md#181-general)), it is *not* part of the interface’s contract.
+
+See §interface-static-constructors for information regarding the allocation and initialization of constants.
+
+> *Note*: See §interface-fields for an example of using various kinds of static members declared within an interface. *end note*
+
+### §interface-fields Interface fields
+
+[§15.5](classes.md#155-fields) and its subclauses cover field declarations in classes. When declaring fields in interfaces, that text needs to be interpreted in the context of an interface, and, where necessary, it is augmented and/or replaced by text in this subclause.
+
+*field_declaration* is described in [§15.5.1](classes.md#1551-general).
+
+It is a compile-time error for *field_declaration* to declare an instance field.
+
+As a static *field_declaration* is considered to have a default implementation ([§18.1](interfaces.md#181-general)), it is *not* part of the interface’s contract.
+
+> *Example*: The following program contains static members of various kinds:
+>
+> <!-- Example: {template:"standalone-console", name:"InterfaceFields", inferOutput:true} -->
+> ```csharp
+> public interface IX
+> {
+>     public const int constant = 100;
+>     protected static int field;
+> 
+>     static IX()
+>     {
+>         Console.WriteLine("static members initialized");
+>         Console.WriteLine($"constant = {IX.constant}, field = {IX.field}");
+>         field = 50;
+>         Console.WriteLine("static constructor has run");
+>     }
+> }
+>
+> public class Test: IX
+> {
+>     public static void Main()
+>     {
+>         Console.WriteLine($"constant = {IX.constant}, field = {IX.field}");
+>     }
+> }
+> ```
+>
+> The output produced is
+>
+> ```console
+> static members initialized
+> constant = 100, field = 0
+> static constructor has run
+> constant = 100, field = 50
+> ```
+>
+> *end example*
+
+See §interface-static-constructors for information regarding the allocation and initialization of static fields.
+
 ### 18.4.2 Interface methods
 
-Interface methods are declared using *interface_method_declaration*s:
+[§15.6](classes.md#156-methods) and its subclauses cover method declarations in classes. When declaring methods in interfaces, that text needs to be interpreted in the context of an interface, and, where necessary, it is augmented and/or replaced by text in this subclause.
 
 ```ANTLR
-interface_method_declaration
-    : attributes? 'new'? return_type interface_method_header
-    | attributes? 'new'? ref_kind ref_return_type interface_method_header
-    ;
-
 interface_method_header
     : identifier '(' parameter_list? ')' ';'
     | identifier type_parameter_list '(' parameter_list? ')'
       type_parameter_constraints_clause* ';'
     ;
```

Comment:
Why retain `interface_method_header` when `interface_method_declaration` is removed?